<launch>
  <param name="robot_type" value="j2s7s300" />
  <param name="robot_connected" value="false" />
  
  <param name="/pick_place_demo/arm/solve_type" value="Manipulation2" />
  <param name="/pick_place_demo/arm/kinematics_solver_timeout" value="0.05" />

  <!-- By default, we are not in debug mode -->
  <arg name="debug" default="false" />

  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <include file="$(find j2s7s300_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
  <include file="$(find j2s7s300_moveit_config)/launch/move_group_j2s7s300.launch">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="false"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="joint_states_ns" value="/j2s7s300/joint_states"/>
    <arg name="controller_manager" value="j2s7s300_ros_control"/>
  </include>

  <!-- ros-control Controller parameters-->
  <rosparam file="$(find j2s7s300_moveit_config)/config/controllers_ros_control.yaml"/>

  <!-- We do not have a robot connected, so publish fake joint states -->
  <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="/use_gui" value="false"/>
    <rosparam param="/source_list">[/move_group/fake_controller_joint_states]</rosparam>
                <param name="zeros/j2s7s300_joint_2" value="3.1415"/>
          <param name="zeros/j2s7s300_joint_4" value="3.1415"/>
    <param name="zeros/j2s7s300_joint_6" value="3.1415"/>
  </node> -->

  <!-- Given the published joint states, publish tf for the robot links -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen">
    <remap from="joint_states" to="/j2s7s300/joint_states" />
  </node>
</launch>